src/types.ts

```diff
@@ -0,0 +1,45 @@
+import { HierarchyRectangularNode } from "d3";
+
+/**
+ * Implied properties for each JSON object in `hierarchy.json`
+ */
+export type Datum = {
+  name: string;
+  value?: number;
+  label?: string;
+  children?: Datum[];
+};
+
+/**
+ * Properties extended by {@link HierarchyRectangularNode}
+ */
+export type Rectangle = {
+  /**
+   * The left edge of the rectangle.
+   */
+  x0: number;
+
+  /**
+   * The top edge of the rectangle
+   */
+  y0: number;
+
+  /**
+   * The right edge of the rectangle.
+   */
+  x1: number;
+
+  /**
+   * The bottom edge of the rectangle.
+   */
+  y1: number;
+};
+
+/**
+ * Stores current and target locations for interactions
+ * @todo Verify this is the correct polymorphism (Should {@link DatumNode} and {@link Rectangle} should be somewhat interchangeable?)
```

Comment:
This inheritance feels wrong, e.g. `shouldBeVisible` currently accepts a `Rectangle` but the initial demo implied a `Datum`

Can ignore for now...